src/fieldnorm/writer.rs

```diff
@@ -6,13 +6,20 @@ use schema::Field;
 use schema::Schema;
 use std::io;
 
+/// The `FieldNormsWriter` is in charge of tracking the fieldnorm byte
+/// of each document for each field with field norms.
+///
+/// `FieldNormsWriter` stores a Vec<u8> for each tracked field, using a
+/// byte per document per field.
 pub struct FieldNormsWriter {
     fields: Vec<Field>,
     fieldnorms_buffer: Vec<Vec<u8>>,
 }
 
 impl FieldNormsWriter {
-    pub fn fields_with_fieldnorm(schema: &Schema) -> Vec<Field> {
+    /// Returns the fields that should have field norms computed
+    /// according to the given schema.
+    pub(crate) fn fields_with_fieldnorm(schema: &Schema) -> Vec<Field> {
```

Comment:
Just want to draw attention to the `pub(crate)` bit here @fulmicoton

Reply:
good idea.